### PR TITLE
Act as an OIDC provider

### PR DESCRIPTION
**Pull request by patrislav**
Depends on #49 

This PR adds support for the `getIdToken` intent that generates an ID token signed by the enclave's KMS signing key.

### CODE REVIEW

File: rpc/identity_provider.go [hunk 1]
```diff
@@ -54,3 +67,91 @@ func (s *RPC) handleJWKS(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	json.NewEncoder(w).Encode(pkd)
 }
+
+func (s *RPC) getIDToken(
+	ctx context.Context, sess *data.Session, intent *intents.IntentTyped[intents.IntentDataGetIdToken],
+) (*intents.IntentResponseIdToken, error) {
+	tnt := tenant.FromContext(ctx)
```
Comment (VojtechVitek) [inline rpc/identity_provider.go:74]: TNT 💣 💥

File: rpc/identity_provider.go [hunk 2]
```diff
@@ -54,3 +67,91 @@ func (s *RPC) handleJWKS(w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	sessData, _, err := crypto.DecryptData[*proto.SessionData](ctx, sess.EncryptedKey, sess.Ciphertext, tnt.KMSKeys)
+	if err != nil {
+		return nil, err
+	}
+
+	var identity proto.Identity
+	if err := identity.FromString(sessData.Identity); err != nil {
+		return nil, fmt.Errorf("parsing session identity: %w", err)
```
Comment (VojtechVitek) [inline rpc/identity_provider.go:87]: I like using present continuous for error messages :)

I read it as "return error parsing session identity", which reads so well 👍

File: rpc/identity_provider.go [hunk 3]
```diff
@@ -54,3 +67,91 @@ func (s *RPC) handleJWKS(w http.ResponseWriter, r *http.Request) {
...
+
+	if acc.Email != "" {
+		b.Claim("email", acc.Email)
+	}
+
+	if intent.Data.Nonce != "" {
+		b.Claim("nonce", intent.Data.Nonce)
+	}
+
+	tok, err := b.Build()
```
Comment (VojtechVitek) [inline rpc/identity_provider.go:124]: minor nit: `token` vs `tok` 

imho, saving few characters doesn't improve readability very much

I'd go for
- acc -> account
- t -> token
- h -> headers
- b -> jwt / jwtBuilder
- mh -> data / b / buf
- tnt -> tenant

File: rpc/identity_provider.go [hunk 4]
```diff
@@ -54,3 +67,91 @@ func (s *RPC) handleJWKS(w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	var identity proto.Identity
+	if err := identity.FromString(sessData.Identity); err != nil {
+		return nil, fmt.Errorf("parsing session identity: %w", err)
+	}
+
+	acc, found, err := s.Accounts.Get(ctx, tnt.ProjectID, identity)
+	if err != nil {
+		return nil, err
```
Comment (VojtechVitek) [inline rpc/identity_provider.go:92]:
```suggestion
		return nil, fmt.Errorf("getting account: %w", err)
```

**Review by VojtechVitek — APPROVED** [review]
LGTM. Nice work :)

Very minor (opinionated) nits.

File: rpc/identity_provider.go (hunk 3 above)
Comment (patrislav) [reply rpc/identity_provider.go:124]: good suggestions, thanks! though in some cases (most notably `tnt`/`tenant`) it is required to not shadow the package name

**Review by VojtechVitek — APPROVED** [review]
LGTM

**Pull request merged**